forge-gui-desktop/src/main/java/forge/screens/home/VLobby.java

```diff
@@ -561,75 +449,75 @@ private void selectMainDeck(final int playerIndex) {
         final Collection<DeckProxy> selectedDecks = mainChooser.getLstDecks().getSelectedItems();
         if (playerIndex < activePlayersNum && lobby.mayEdit(playerIndex)) {
             final String text = type.toString() + ": " + Lang.joinHomogenous(selectedDecks, DeckProxy.FN_GET_NAME);
-            playerPanels.get(playerIndex).setDeckSelectorButtonText(text);
+            getPlayerPanel(playerIndex).setDeckSelectorButtonText(text);
             fireDeckChangeListener(playerIndex, deck);
         }
         mainChooser.saveState();
     }
 
-    private void selectCommanderDeck(final int playerIndex) {
+    private FDeckChooser getDeckChooser(final int iSlot) {
+        return getPlayerPanel(iSlot).getDeckChooser();
+    }
+
+    private void selectCommanderDeck(final FDeckChooser mainChooser, final int playerIndex) {
         if (!hasVariant(GameType.Commander) && !hasVariant(GameType.Oathbreaker) && !hasVariant(GameType.TinyLeaders) && !hasVariant(GameType.Brawl)) {
             // Only these game types use this specific deck panel
             return;
         }
-        final FDeckChooser mainChooser = getCommanderDeckChooser(playerIndex);
         final DeckType type = mainChooser.getSelectedDeckType();
```

Comment:
It seems like these 4 functions also share almost all of their code that could be refactored/merged
(not sure if these sanity checks are actually still useful)

Reply:
True! Done.